federation/pkg/kubefed/init/init.go

```diff
@@ -173,6 +176,7 @@ func (o *initFederationOptions) Bind(flags *pflag.FlagSet, defaultServerImage, d
 	flags.Int32Var(&o.apiServerNodePortPort, apiserverPortFlag, 0, "Preferred port to use for api server nodeport service (0 for random port assignment). Valid only if '"+apiserverServiceTypeFlag+"=NodePort'.")
 	flags.BoolVar(&o.apiServerEnableHTTPBasicAuth, "apiserver-enable-basic-auth", false, "Enables HTTP Basic authentication for the federation-apiserver. Defaults to false.")
 	flags.BoolVar(&o.apiServerEnableTokenAuth, "apiserver-enable-token-auth", false, "Enables token authentication for the federation-apiserver. Defaults to false.")
+	flags.StringVar(&o.tolerationsString, "tolerations", "", "If specified, the pod's tolerations. Example \"key1=value1:NoSchedule,key2=value2:NoSchedule\"")
```

Comment:
@gyliu513 By template, I meant something similar but slightly different than #50817.  As I understand it, #50817 proposes allowing the use of a file containing parameters *instead of* a bunch of command line arguments. What I was proposing above was allowing a template file to be specified, into which the other command line parameters would be substituted (by kubefed).  So externalizing the hard-coded template around line 706+ in init.go below into a file that the user could customize.